distributed/tests/test_scheduler.py

```diff
@@ -194,6 +194,39 @@ def test_remove_worker_from_scheduler(s, a, b):
     s.validate_state()
 
 
+@gen_cluster(config={"distributed.scheduler.events-cleanup-delay": 0.01})
+def test_clear_events_worker_removal(s, a, b):
+    assert a.address in s.events
+    assert a.address in s.ncores
+    assert b.address in s.events
+    assert b.address in s.ncores
+
+    s.remove_worker(address=a.address)
+    # Shortly after removal, the events should still be there
+    assert a.address in s.events
+    assert a.address not in s.ncores
+    s.validate_state()
+
+    yield gen.sleep(0.02)
```

Comment:
The time here is short enough that it might fail intermittently.  This is particularly a problem on Travis-CI, which can be *very* slow sometimes.  Instead, I recommend the following pattern:

```python
start = time()
while a.address in s.events:
    yield gen.sleep(0.01)
    assert time() < start + 2
assert b.address in s.events
```

This says "this condition should happen soonish, but we don't mind if it's a second or two late" (which could be something like a `gc.collect` in a bad case.